sparse/impl/KokkosSparse_sptrsv_cuSPARSE_impl.hpp

```diff
@@ -60,6 +60,142 @@ void sptrsvcuSPARSE_symbolic(KernelHandle* sptrsv_handle,
+               ? (void*)sptrsv_handle->get_int64_rowmap_ptr_copy(row_map)
+               : (void*)row_map.data();
+    }
+    const scalar_type alpha = scalar_type(1.0);
+
+    cusparseIndexType_t cudaCsrColIndType, cudaCsrRowMapType;
+    cudaDataType cudaValueType;
+
+    if (std::is_same<idx_type, int>::value) {
+      cudaCsrColIndType = CUSPARSE_INDEX_32I;
+      cudaCsrRowMapType = CUSPARSE_INDEX_32I;
+    } else {  // idx_type has 64 bits
+      cudaCsrColIndType = CUSPARSE_INDEX_64I;
+      cudaCsrRowMapType = CUSPARSE_INDEX_64I;
+    }
+
+    if (std::is_same<scalar_type, double>::value) {
+      cudaValueType = CUDA_R_64F;
+    } else if (std::is_same<scalar_type, float>::value) {
+      cudaValueType = CUDA_R_32F;
+    } else if (std::is_same<scalar_type, Kokkos::complex<double>>::value) {
+      cudaValueType = CUDA_C_64F;
+    } else if (std::is_same<scalar_type, Kokkos::complex<float>>::value) {
+      cudaValueType = CUDA_C_32F;
+    } else {
+      throw std::runtime_error("CUSPARSE wrapper error: unsupported type.\n");
+    }
+
+    // Create sparse matrix in CSR format
+    status = cusparseCreateCsr(&(h->matDescr), static_cast<int64_t>(nrows),
+                               static_cast<int64_t>(nrows), nnz, rm,
+                               (void*)entries.data(), (void*)values.data(),
+                               cudaCsrRowMapType, cudaCsrColIndType,
+                               CUSPARSE_INDEX_BASE_ZERO, cudaValueType);
+
+    // Create dummy dense vector B (RHS)
+    nnz_scalar_view_t b_dummy("b_dummy", nrows);
+    cusparseCreateDnVec(&(h->vecBDescr_dummy), static_cast<int64_t>(nrows),
+                        b_dummy.data(), cudaValueType);
+
+    // Create dummy dense vector X (LHS)
+    nnz_scalar_view_t x_dummy("x_dummy", nrows);
+    cusparseCreateDnVec(&(h->vecXDescr_dummy), static_cast<int64_t>(nrows),
+                        x_dummy.data(), cudaValueType);
+
+    // Specify Lower|Upper fill mode
+    if (is_lower) {
+      cusparseFillMode_t fillmode = CUSPARSE_FILL_MODE_LOWER;
+      cusparseSpMatSetAttribute(h->matDescr, CUSPARSE_SPMAT_FILL_MODE,
+                                &fillmode, sizeof(fillmode));
+    } else {
+      cusparseFillMode_t fillmode = CUSPARSE_FILL_MODE_UPPER;
+      cusparseSpMatSetAttribute(h->matDescr, CUSPARSE_SPMAT_FILL_MODE,
+                                &fillmode, sizeof(fillmode));
+    }
+
+    // Specify Unit|Non-Unit diagonal type.
+    cusparseDiagType_t diagtype = CUSPARSE_DIAG_TYPE_NON_UNIT;
+    cusparseSpMatSetAttribute(h->matDescr, CUSPARSE_SPMAT_DIAG_TYPE, &diagtype,
+                              sizeof(diagtype));
+
+    // Allocate an external buffer for analysis
+    status = cusparseSpSV_bufferSize(
+        h->handle, h->transpose, &alpha, h->matDescr, h->vecBDescr_dummy,
+        h->vecXDescr_dummy, cudaValueType, CUSPARSE_SPSV_ALG_DEFAULT,
+        h->spsvDescr, &pBufferSize);
+
+    if (CUSPARSE_STATUS_SUCCESS != status)
```

Comment:
If you wrapped the previous call you do not need to check, at this point you might have overwritten the initial error and not really know what happened anymore : (

Reply:
Removed.